### PR TITLE
chore(AppEngine): update appengine reference to match new schema

### LINKED ISSUE #41: Update AppEngine Reference doc to match UI refactor

### PR DESCRIPTION
**Pull request by conneryn**
## Description

Update AppEngine reference documentation to match new schema.

Fixes #41

### CODE REVIEW

File: content/references/appengine.md [hunk 1]
```diff
@@ -186,3 +198,189 @@ editions:
...
+### Prompt Extensions
+
+| Property  | Value(s) | Default  | Description
+| --------  | -------- | -------  | -----------
+| target    | String   | `configs`| Where should the variable be stored?  Valid values: `configs`, `secrets`, `transient`.
+| label     | String   |          | An additional label use when displaying the input field.
+| required  | Boolean  | `false`  | Require a response in order to proceed.
+| generate  | [Generator](#Generator) |   | Provide the user an option to auto-generate a value.
+| generateMessage | String |      | Message to the user regarding the auto-generation of the value.
+| value     | String    |         |
```
Comment (conneryn) [inline content/references/appengine.md:330]: @nothingmn or @nsainaney How does `value` differ from the inquirer's `default` value?

File: content/references/appengine.md [hunk 2]
```diff
@@ -186,3 +198,189 @@ editions:
...
+        - title: Section 2
+          prompts: 
+          - name: LAST_NAME
+            message: What is your last name?
+```
+### Prompt
+
+| Property  | Value(s) | Default  | Description
+| --------  | -------- | -------  | -----------
+| type      | String   | `input`    | Supported types: `input`, `number`, `confirm`, `list`, `rawlist`, `expand`, `checkbox`, `password`, or `editor`.
```
Comment (conneryn) [inline content/references/appengine.md:288]: @nsainaney @nothingmn I'm assuming we dont' support all of these yet, so which types should I actually list here?

File: content/references/appengine.md [hunk 3]
```diff
@@ -186,3 +198,189 @@ editions:
...
+| --------  | -------- | -------  | -----------
+| target    | String   | `configs`| Where should the variable be stored?  Valid values: `configs`, `secrets`, `transient`.
+| label     | String   |          | An additional label use when displaying the input field.
+| required  | Boolean  | `false`  | Require a response in order to proceed.
+| generate  | [Generator](#Generator) |   | Provide the user an option to auto-generate a value.
+| generateMessage | String |      | Message to the user regarding the auto-generation of the value.
+| value     | String    |         | 
+| maxlength | Int       |         | Maximum length allowed.
+| min       | Int       | `1`     | Require value to be larger than min value (when prompt type is `number`).
+| max       | Int       | `32767` | Require value to be less than max value (when prompt type is `number`).
```
Comment (conneryn) [inline content/references/appengine.md:337]: Do these validation rules (max, min, maxlength, required), also affect the CLI inquire stages, or just the WebUI?

File: content/references/appengine.md [hunk 4]
```diff
@@ -186,3 +198,189 @@ editions:
...
+
+| Property  | Value(s) | Default  | Description
+| --------  | -------- | -------  | -----------
+| target    | String   | `configs`| Where should the variable be stored?  Valid values: `configs`, `secrets`, `transient`.
+| label     | String   |          | An additional label use when displaying the input field.
+| required  | Boolean  | `false`  | Require a response in order to proceed.
+| generate  | [Generator](#Generator) |   | Provide the user an option to auto-generate a value.
+| generateMessage | String |      | Message to the user regarding the auto-generation of the value.
+| value     | String    |         | 
+| maxlength | Int       |         | Maximum length allowed.
```
Comment (conneryn) [inline content/references/appengine.md:335]: Just want to confirm that this is all lowercase `maxlength` (rather than with a capital `L`).

**Review by nsainaney — DISMISSED** [review]

**Pull request merged**